src/model/database.py

```diff
@@ -20,10 +20,36 @@ def serialize(self):
         self.__serialize_tables__()
 
     @staticmethod
-    def __validate__(schema_data):
-        if len(schema_data[Keys.DATABASE]) == 0 or schema_data[Keys.DATABASE].isspace():
+    def __validate__(schema_data, database_name):
+        if (len(schema_data[Keys.DATABASE]) == 0 or schema_data[Keys.DATABASE].isspace()) and (database_name is None or str(database_name).isspace()):
             raise WrongParameterError("No database detected")
 
     def __serialize_tables__(self):
         for table in self.tables:
             table.serialize()
+
+    def __initialize_by_schema_data__(self, schema_data):
+        self.__path__ = os.path.join(self.DATABASE_PATH, schema_data[Keys.DATABASE])
+        self.__database_name__ = schema_data[Keys.DATABASE]
+        self.tables = {}
+        for table in schema_data[Keys.TABLES]:
+            self.tables.__setattr__(str(table[Keys.NAME]), Table(self, str(table[Keys.NAME]), table))
+
+    def __initialize_by_database_name(self, database_name):
+        self.__path__ = os.path.join(Database.DATABASE_PATH, database_name)
+        if not os.path.exists(self.__path__):
+            raise WrongParameterError("Wrong database entered")
+        self.__database_name__ = database_name
+        self.tables = {}
+        for table in os.listdir(path):
+            self.tables.__setattr__(str(table), Table(self, str(table)))
+
+    def get_table(self, table_name):
+        for table in self.tables:
+            if table.get_name() == table_name:
+                return table
+        raise WrongParameterError("Wrong table entered")
```

Comment:
again, you didn't save the table in the `Dic`. please use Dic in constructor instead of array